src/post/ExperimentalPostSchema.ts

```diff
@@ -0,0 +1,51 @@
+import { z } from 'zod';
+
+import { NonEmptyStringSchema, type Signature } from '../primitives.js';
+import type { NftMetadata } from '../tokens/eip721.js';
+import { PostMainFocus } from './PostMainFocus.js';
+import { PostMetadataSchemaId } from './PostMetadataSchemaId.js';
+import { type PostMetadataCommon, metadataDetailsWith, postWith } from './common/index.js';
+
+export type ExperimentalPostMetadataDetails = PostMetadataCommon & {
+  /**
+   * The main focus of the post.
+   */
+  mainContentFocus: PostMainFocus;
+  /**
+   * A sub-identifier for the custom post type.
+   * This is used to differentiate between different custom post types.
+   */
+  name: string;
+};
+
+const ExperimentalPostMetadataDetailsSchema: z.ZodType<
+  ExperimentalPostMetadataDetails,
+  z.ZodTypeDef,
+  object
+> = metadataDetailsWith({
+  mainContentFocus: z.nativeEnum(PostMainFocus),
+
+  name: NonEmptyStringSchema.describe(
+    'A sub-identifier for the custom post type. This is used to differentiate between different custom post types.',
+  ),
```

Comment:
What used to be value (a JSON string) should be done with attributes in the most typed way if possible (at first) with JSONString attribute as last resort.